packages/fuels-types/src/bech32.rs

```diff
@@ -114,32 +137,108 @@ mod test {
+        let pubkey_hashes = [
+            [
+                107, 50, 223, 89, 84, 225, 186, 222, 175, 254, 253, 44, 15, 197, 229, 148, 220,
+                255, 55, 19, 170, 227, 221, 24, 183, 217, 102, 98, 75, 1, 0, 39,
+            ],
+            [
+                49, 83, 18, 64, 150, 242, 119, 146, 83, 184, 84, 96, 160, 212, 110, 69, 81, 34,
+                101, 86, 182, 99, 62, 68, 44, 28, 40, 26, 131, 21, 221, 64,
+            ],
+            [
+                48, 101, 49, 52, 48, 102, 48, 55, 48, 100, 49, 97, 102, 117, 51, 57, 49, 50, 48,
+                54, 48, 98, 48, 100, 48, 56, 49, 53, 48, 52, 49, 52,
+            ],
         ];
-
-        let bech32_contract_id = &Bech32ContractId::from_str(
+        let bech32m_encodings = [
+            "fuel1dved7k25uxadatl7l5kql309jnw07dcn4t3a6x9hm9nxyjcpqqns50p7n2",
+            "fuel1x9f3ysyk7fmey5ac23s2p4rwg4gjye2kke3nu3pvrs5p4qc4m4qqwx56k3",
             "fuel1xpjnzdpsvccrwvryx9skvafn8ycnyvpkxp3rqeps8qcn2vp5xy6qu7yyz7",
-        )
-        .unwrap();
-        let plain_contract_id: ContractId = bech32_contract_id.into();
+        ];
+        let hex_encodings = [
+            "0x6b32df5954e1badeaffefd2c0fc5e594dcff3713aae3dd18b7d966624b010027",
+            "0x3153124096f2779253b85460a0d46e4551226556b6633e442c1c281a8315dd40",
+            "0x3065313430663037306431616675333931323036306230643038313530343134 ",
+        ];
+
+        {
+            for i in 0..3 {
+                let bech32_contract_id = &Bech32ContractId::from_str(bech32m_encodings[i]).unwrap();
+                assert_eq!(*bech32_contract_id.hash(), pubkey_hashes[i]);
+            }
+
+            for i in 0..3 {
+                let bech32_contract_id = &Bech32ContractId::from_str(hex_encodings[i]).unwrap();
+                assert_eq!(*bech32_contract_id.hash(), pubkey_hashes[i]);
+            }
+        }
+
+        {
+            for i in 0..3 {
+                let bech32_contract_id = &Bech32Address::from_str(bech32m_encodings[i]).unwrap();
+                assert_eq!(*bech32_contract_id.hash(), pubkey_hashes[i]);
+            }
 
-        let expected_contract_id = ContractId::new(pubkey_hash);
+            for i in 0..3 {
+                let bech32_contract_id = &Bech32Address::from_str(hex_encodings[i]).unwrap();
+                assert_eq!(*bech32_contract_id.hash(), pubkey_hashes[i]);
+            }
+        }
+    }
 
-        assert_eq!(plain_contract_id, expected_contract_id);
+    #[test]
+    fn test_from_invalid_bech32_string() {
+        {
+            let expected: Error = bech32::Error::InvalidChecksum.into();
+            let invalid_bech32_with_hex_chars =
+                "fadeaffe1dfed7c25adadadd715e3f309ecf07dca4e3a6d9de9cdadcddac50d7f";
+
+            {
+                let result = &Bech32ContractId::from_str(invalid_bech32_with_hex_chars)
+                    .expect_err("should error");
+                assert_eq!(result.to_string(), expected.to_string());
```

Comment:
Here I think you should check for the error, even though IIUC, it's going to be `GeneralError` which is very broad.